### PR TITLE
Read EM processes from G4GammaGeneral

### LINKED ISSUE #683: Support importing G4GammaGeneralProcess
The G4GammaGeneralProcess is a performance enhancement for Geant4 that uses a single "total" cross section (and sampling during a collision) instead of independently sampling several different path lengths for processes. (Celeritas does this by default.) Unfortunately this means the G4VProcess physics table has only a single physics table for cross section instead of one per process, which we need to build our own processes. There may be a way we can reconstruct our needed processes from the gamma general?

### PR DESCRIPTION
**Pull request by sethrj**
This fixes #683 by refactoring the process importer and processing a single G4GammaGeneralProcess into several ImportProcess instances. The key addition is 2fb92949191287cca1b5c06aaa192e10b1e16d5c .

### CODE REVIEW

**Comment by sethrj** [comment]
@whokion I think this should fix another one of the CMS problems. Could you try it out with the framework this week?

File: src/celeritas/ext/detail/GeantPhysicsList.cc [hunk 1]
```diff
@@ -128,31 +130,47 @@ void GeantPhysicsList::add_gamma_processes()
     auto* physics_list = G4PhysicsListHelper::GetPhysicsListHelper();
     auto* gamma = G4Gamma::Gamma();
 
+    // Option to create GammaGeneral for performance/robustness
```
Comment (stognini) [inline src/celeritas/ext/detail/GeantPhysicsList.cc:138]: Although the lambda function is straightforward to read, I think it's worth adding a note before or after the table in the documentation of `add_gamma_process()` mentioning that we either load individual processes or gamma general, which represents all processes in the table.

**Comment by stognini** [comment]
Great work. Didn't find any problems running some checks beyond ctest. I like the refactoring, especially removing the temporary `GeantProcessImporter::process_` and simplifying import by overloading `operator()`. Since `G4GammaGeneralProcess` is technically a wrapper, do you think we'd benefit from having an extra test harness for that? (I.e. verifying that by importing the data using `GeantPhysicsOptions::gamma_general = true` we end up with the same gamma processes as if we'd loaded them manually.)

**Review by stognini — APPROVED** [review]

**Comment by sethrj** [comment]
Great, thanks! I'll update the documentation and then merge after @whokion checks out that this works for CMS.

As to a test: the new `OneSteelSphereGG` in GeantImporter.test.cc builds the G4GammaGeneral process and checks that the loaded processes are the same :) (That test found a bug or two in my implementation...) Were you thinking of something more?

**Comment by whokion** [comment]
@sethrj Tested this branch with CMSSW, which works fine.  One remaining warning besides the known warning
```
celeritas/src/celeritas/phys/ProcessBuilder.cc:217: warning: Omitting coulomb_scat from physics process list
```
 is
```
celeritas/ext/detail/GeantModelImporter.cc:130: warning: Encountered unknown process 'PhotoElectric
```
 which seems due to the difference in naming for the `photoelectric` process as we names it as "LivermorePhElectric".

**Comment by sethrj** [comment]
@whokion Good news for the first half, and the second half points to a bug in the output: `PhotoElectric` corresponds to the `G4PEEffectFluoModel` model, not a process as the output erroneously said. Since we don't do anything with the model class information at this point, that error can be ignored, but I'll add it to this PR too. Thanks for checking!

**Pull request merged**